deku-p/src/core/block_storage/tests/block_storage_tests.ml

```diff
@@ -0,0 +1,68 @@
+open Deku_stdlib
+open Deku_crypto
+open Deku_consensus
+open Deku_concepts
+
+let block_testable = Alcotest.testable Block.pp Block.equal
+
+let identity =
+  let secret = Ed25519.Secret.generate () in
+  let secret = Secret.Ed25519 secret in
+  Identity.make secret
+
+let block ~default_block_size =
+  let above = Genesis.block in
+  let withdrawal_handles_hash = BLAKE2b.hash "potato" in
+  let producer = Producer.empty in
+  Producer.produce ~identity ~default_block_size ~above ~withdrawal_handles_hash
+    producer
+
+let file_hash =
+  let randn = Stdlib.Random.int 230 in
+  Deku_crypto.BLAKE2b.hash (Int.to_string randn) |> BLAKE2b.to_hex
+
+let uri = Uri.of_string (Format.sprintf "sqlite3:/tmp/%s.db" file_hash)
+
+let make_block_storage env sw =
+  let domains = Eio.Stdenv.domain_mgr env in
+  let worker = Parallel.Worker.make ~domains ~sw in
+  let storage = Deku_block_storage.Block_storage.make ~worker ~uri in
+  storage
+
+let test_empty_block_load env () =
+  Eio.Switch.run @@ fun sw ->
+  let block_storage = make_block_storage env sw in
+  let (Block { hash; _ } as block) = block ~default_block_size:0 in
+  Deku_block_storage.Block_storage.save_block ~block block_storage;
+  let retrieved_block =
+    match
+      Deku_block_storage.Block_storage.find_block_by_hash ~block_hash:hash
+        block_storage
+    with
+    | Some json -> Block.t_of_yojson json
+    | None -> Genesis.block
+  in
+  Alcotest.(check' block_testable)
+    ~msg:"hash loaded block is not equal to saved block" ~expected:block
+    ~actual:retrieved_block;
+
+  let (Deku_block_storage.Block_storage.Storage { worker; _ }) =
+    block_storage
+  in
+  Parallel.Worker.teardown worker
+
+let eio_test_case : (Eio.Stdenv.t -> unit -> unit) -> unit -> unit =
+ fun f () -> Eio_main.run (fun env -> f env ())
```

Comment:
you should never run Eio_main.run many times.

Reply:
Ok. As it's not a bug, and matters for performance, I'll address this in a future PR.